### PR TITLE
upgrade terraform-plugin-sdk

### PR DESCRIPTION
**Pull request by juliolustosa**
fix bugs and work with terraform 1.3.x

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -38,4 +38,5 @@ debug_test:
 	TF_LOG=debug make test
 
 generate-docs:
+	go get github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs
```
Comment (tcarreira) [inline GNUmakefile]: should be 
```yaml
go install github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs@0.13.0
```


from https://tip.golang.org/doc/go1.16#modules :
> This is useful for installing executables without affecting the dependencies of the main module.

File: go.mod [hunk 2]
```diff
@@ -1,9 +1,84 @@
 module github.com/tsuru/terraform-provider-hcaas
 
-go 1.16
+go 1.18
 
 require (
-	github.com/hashicorp/terraform-plugin-docs v0.5.0 // indirect
-	github.com/hashicorp/terraform-plugin-sdk/v2 v2.8.0
+	github.com/hashicorp/terraform-plugin-sdk/v2 v2.23.0
```
Comment (tcarreira) [inline go.mod:6]: Would you please consider 2 different commits:
- one for upgrading go
- another for upgrading terraform sdk?

It's easier to maintain and pinpoint any issue in the future.

**Review by tcarreira — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (juliolustosa) [reply go.mod:6]: não é possível atualizar o SDK sem atualizar o go é incompatível

File: GNUmakefile (hunk 1 above)
Comment (juliolustosa) [reply GNUmakefile]: change

**Review by tcarreira — APPROVED** [review]

**Pull request merged**